week-5.c

```diff
@@ -0,0 +1,138 @@
+
+#include <stdio.h>
+#include<stdlib.h>
+
+
+struct TreeNode
+{	
+    int value;
+    struct TreeNode* left;
+    struct TreeNode* right;
+
+
+};
+
+struct TreeNode* newNode(int data)
+{
+    struct TreeNode* node = (struct TreeNode*)
+                             malloc(sizeof(struct TreeNode));
+    node->value  = data;
+    node->left  = NULL;
+    node->right = NULL;
+ 
+    return(node);
+};
+
+struct TreeNode *root1=NULL;
+struct TreeNode *root2=NULL;
+int flag=0;  //// to keep a track if not a single subtree of 1st tree matches with 2nd tree 
+
+
+int Is_Same(struct TreeNode* root1, struct TreeNode* root2) ///// check if 2 trees are identical or not
+{
+    /* both empty */
+    if (root1==NULL && root2==NULL)
+        return 1;
+ 
+    /* both non-empty */
+    if (root1!=NULL && root2!=NULL)
+    {
+        
+	if (root1->value == root2->value)
+        {
+            if( Is_Same(root1->left, root2->left) && Is_Same(root1->right, root2->right))
+		return 1;
+        }
+	else return 0;
+
+    }
+     
+    /*  one empty, one not  */
+    return 0;
+} 
+
+
+///// inorder traversing of 1st tree to check if 2nd tree matches the subtree of 1st tree with the current node as the root of the subtree 
+
+void check_inorder(struct TreeNode * tree)
+{
+    if (tree)
+    {
+        check_inorder(tree->left);
+	
+	if(Is_Same(tree, root2))
+        {	printf("Both trees are identical at node %d of 1st tree",tree->value);
+		flag=1;
+	}
+   		
+	check_inorder(tree->right);
+    }
+}
+
+
+struct TreeNode* insert(struct TreeNode* node, int value) ///// insertion of an element in a binary tree. I have used the method of insertion used in Binary Search Tree. We can use other methods also.
```

Comment:
As you mention in your comment, the tree doesn't necessarily have to be a binary search tree and can have the data in any arbitrary order.